### PR TITLE
Feat/form validation

### LINKED ISSUE #250: [Feature] Form Validation & Form Component
## Description

**Is your feature request related to a problem? Please describe.**

We should use system for validating our forms.
We don't have any system to check if our form have valid state so some times our forms validation are not right. 

And in the same time we should try to have a form component that reduce the number of different form component we are creating.

## Ressources
https://formkit.com/essentials/validation
https://ui.nuxt.com/components/form

## Acceptance Criteria

```[tasklist]
### Tasks
- [ ] https://github.com/globe-and-citizen/cnc-portal/issues/228
- [x] [Feature] Form Validation sytem
- [ ] [Feature] Form Component
```

### PR DESCRIPTION
**Pull request by dasarathg68**
# Description

Fixes #250 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

* **Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)

### CODE REVIEW

File: app/src/components/forms/DepositBankForm.vue [hunk 1]
```diff
@@ -19,10 +20,29 @@
...
+import { required, numeric } from '@vuelidate/validators'
+import { useVuelidate } from '@vuelidate/core'
 
 const amount = ref<string>('0')
-defineEmits(['deposit', 'closeModal'])
+const emits = defineEmits(['deposit', 'closeModal'])
 defineProps<{
   loading: boolean
 }>()
+
```
Comment (farrellh1) [inline app/src/components/forms/DepositBankForm.vue:31]: can you add not zero validation like this
```typescript
const notZero = helpers.withMessage('Amount must be greater than 0', (value: string) => {
  return parseFloat(value) > 0
})
```

File: app/src/components/forms/TransferFromBankForm.vue [hunk 2]
```diff
@@ -72,9 +87,35 @@ defineProps<{
   bankBalance: string | null
   filteredMembers: User[]
 }>()
+
```
Comment (farrellh1) [inline app/src/components/forms/TransferFromBankForm.vue:90]: also in here add `notZero` validation

Comment (dasarathg68) [reply app/src/components/forms/TransferFromBankForm.vue]: Done, thanks

File: app/src/components/forms/DepositBankForm.vue (hunk 1 above)
Comment (dasarathg68) [reply app/src/components/forms/DepositBankForm.vue]: Done

**Review by hermannleboss — COMMENTED** [review]
Add test for error useCase
Move test files to fit the folder structure.

**Pull request merged**